apps/etterlatte-behandling/src/main/kotlin/oppgave/OppgaveService.kt

```diff
@@ -228,53 +224,85 @@ class OppgaveService(
+                .filter { it.type == type }
+                .filter { it.status in listOf(Status.UNDER_BEHANDLING, Status.UNDERKJENT) }
+
+        if (oppgaver.isEmpty()) {
+            throw ManglerOppgaveUnderBehandling("Ingen oppgave funnet for referanse: $referanse")
+        } else if (oppgaver.size > 1) {
+            throw ForMangeOppgaverUnderBehandling("For mange oppgaver under behandling (ref: $referanse)")
         }
-        try {
-            val oppgaveUnderbehandling = behandlingsoppgaver.single { it.status == Status.UNDER_BEHANDLING }
-            ferdigstillOppgaveById(oppgaveUnderbehandling, saksbehandler)
-            return opprettNyOppgaveMedSakOgReferanse(
-                referanse = fattetoppgaveReferanseOgSak.referanse,
-                sakId = fattetoppgaveReferanseOgSak.sakId,
-                oppgaveKilde = oppgaveUnderbehandling.kilde,
-                oppgaveType = oppgaveType,
-                merknad = merknad,
-            )
-        } catch (e: NoSuchElementException) {
-            throw BadRequestException(
-                "Det må finnes en oppgave under behandling, gjelder behandling:" +
-                    " ${fattetoppgaveReferanseOgSak.referanse}",
-                e,
-            )
-        } catch (e: IllegalArgumentException) {
-            throw BadRequestException(
-                "Skal kun ha en oppgave under behandling, gjelder behandling:" +
-                    " ${fattetoppgaveReferanseOgSak.referanse}",
-                e,
-            )
+
+        val oppgave = oppgaver.single()
+
+        val oppdatertMerknad = merknad ?: oppgave.merknad ?: ""
+        oppgaveDao.oppdaterStatusOgMerknad(oppgave.id, oppdatertMerknad, Status.ATTESTERING)
+
+        if (frist != null) {
+            oppgaveDao.redigerFrist(oppgave.id, frist)
+        }
+        oppgaveDao.fjernSaksbehandler(oppgave.id)
+
+        return hentOppgave(oppgave.id)
+    }
+
+    fun tilUnderkjent(
+        referanse: String,
+        type: OppgaveType,
+        merknad: String?,
+    ): OppgaveIntern {
+        val oppgave =
+            hentOppgaverForReferanse(referanse)
+                .filter { it.type == type }
+                .singleOrNull { it.erAttestering() }
+                ?: throw IllegalStateException("Fant ikke oppgave med referanse: $referanse")
+
+        val oppdatertMerknad = merknad ?: oppgave.merknad ?: ""
+        oppgaveDao.oppdaterStatusOgMerknad(oppgave.id, oppdatertMerknad, Status.UNDERKJENT)
+
+        val saksbehandler =
+            oppgaveDao.hentEndringerForOppgave(oppgave.id)
+                .sortedByDescending { it.tidspunkt }
+                .firstOrNull { it.oppgaveEtter.erAttestering() }
+                ?.oppgaveFoer
+                ?.saksbehandler
+
+        if (saksbehandler != null) {
+            oppgaveDao.settNySaksbehandler(oppgave.id, saksbehandler.ident)
+        } else {
+            logger.warn("Fant ikke siste saksbehandler for oppgave med referanse: $referanse")
```

Comment:
Vi forventer vel en del innslag på denne for alle oppgaver som er til attestering når denne migreringen skjer, men etter de er tatt unna er dette kanskje verdt å logge en error på? Bare en tanke

Reply:
Legger til en TODO! 👍